packages/fast-components-react-base/src/context-menu-item/context-menu-item.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { get } from "lodash-es";
+import { ContextMenuItemClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    ContextMenuItemHandledProps,
+    ContextMenuItemManagedClasses,
+    ContextMenuItemProps,
+    ContextMenuItemUnhandledProps,
+} from "./context-menu-item.props";
+
+export enum ContextMenuItemRole {
+    menuitem = "menuitem",
+    menuitemradio = "menuitemradio",
+    menuitemcheckbox = "menuitemcheckbox",
+}
+
+class ContextMenuItem extends Foundation<
+    ContextMenuItemHandledProps,
+    ContextMenuItemUnhandledProps,
+    {}
+> {
+    public static displayName: string = "ContextMenuItem";
+
+    public static defaultProps: Partial<ContextMenuItemProps> = {
+        role: ContextMenuItemRole.menuitem,
+        disabled: false,
+    };
+
+    protected handledProps: HandledProps<ContextMenuItemHandledProps> = {
+        managedClasses: void 0,
+        children: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                role={this.props.role}
+                aria-disabled={this.props.disabled || undefined}
+            >
+                {this.props.children}
+            </div>
+        );
+    }
+
+    /**
+     * Create class-names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props.managedClasses, "contextMenuItem") || "";
+
+        switch (this.props.role) {
+            case ContextMenuItemRole.menuitemcheckbox:
+                className = className.concat(

```

Comment:
Just wondering why you are doing it this way, as opposed to `className += ` ${get(...class)}`?

Reply:
eh,  I guess I just found it easier to read. I find string interpolation *can* become harder to read then a list of arguments.